pkg/user/appointee/set.go

```diff
@@ -59,29 +63,97 @@ func setAppointeePermission(
 	if err != nil {
 		return err
 	}
-	receipt, err := permissionWriter.SetPermission(
-		ctx,
-		elcontracts.SetPermissionRequest{
-			AccountAddress:   config.AccountAddress,
-			AppointeeAddress: config.AppointeeAddress,
-			Target:           config.Target,
-			Selector:         config.Selector,
-			WaitForReceipt:   true,
-		},
-	)
+	return broadcastOrPrint(ctx, logger, permissionWriter, config)
+}
+
+func broadcastOrPrint(
+	ctx context.Context,
+	logger logging.Logger,
+	permissionWriter SetAppointeePermissionWriter,
+	config *setConfig) error {
+	var err error
+	permissionRequest := elcontracts.SetPermissionRequest{
+		AccountAddress:   config.AccountAddress,
+		AppointeeAddress: config.AppointeeAddress,
+		Target:           config.Target,
+		Selector:         config.Selector,
+		WaitForReceipt:   true,
+	}
+	if config.Broadcast {
+		err = broadcastSetAppointeeCallData(ctx, permissionWriter, config, permissionRequest)
+	} else {
+		err = printSetAppointeeCallData(logger, permissionWriter, config, permissionRequest)
+	}
+	return err
```

Comment:
```suggestion
	if config.Broadcast {
		return broadcastSetAppointeeCallData(ctx, permissionWriter, config, permissionRequest)
	} 
	return printSetAppointeeCallData(logger, permissionWriter, config, permissionRequest)
```

Reply:
Keeping this as is to prefer fewer return statements